data/dataset.go

```diff
@@ -0,0 +1,63 @@
+package data
+
+import (
+	torch "github.com/wangkuiyi/gotorch"
+)
+
+// Example contains data and target
+type Example struct {
+	data, target torch.Tensor
+	hasGCed      bool
+}
+
+// NewExample creates an example from `data` and `target`
+func NewExample(data, target torch.Tensor) *Example {
+	return &Example{data, target, false}
+}
+
+// Data of the example
+func (e *Example) Data() torch.Tensor {
+	if !e.hasGCed {
+		torch.GC()
+		e.hasGCed = true
+	}
+	torch.SetTensorFinalizer(e.data.T)
```

Comment:
By calling `torch.SetTensorFinalizer` in `Example.Data/Target`, could we fix the bug https://github.com/wangkuiyi/gotorch/issues/273 ? @shendiaomo @Yancey1989

Reply:
It seems this PR works with the C++ wrappered dataset, #273 using `ImageLoader` which is Go implementation, it seems this PR can not fix it.